### PR TITLE
Make indoor maps zoomable

### LINKED ISSUE #98: Indoor Maps Zoom
Zoom into indoor maps images
https://medium.com/quick-code/pinch-to-zoom-with-multi-touch-gestures-in-android-d6392e4bf52d

### PR DESCRIPTION
**Pull request by patrickfeltes**
Closes #98 

We should definitely test this on alpha to see how people's phones react to the size of these images.

### CODE REVIEW

File: app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java [hunk 1]
```diff
@@ -46,13 +50,13 @@ public void onTabReselected(TabLayout.Tab tab) {
     private void setImage(int index) {
         switch (index) {
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java]: If you want to save a lot of lines of code, you could make a dictionary with entries like `DCL -> R.drawable.dcl_indoor` and query that with the index

File: app/build.gradle [hunk 2]
```diff
@@ -39,6 +39,9 @@ dependencies {
     implementation 'com.android.support:design:28.0.0'
     implementation 'com.android.support.constraint:constraint-layout:1.1.3'
 
+    // zoomable images
```
Comment (bstrauch24) [inline app/build.gradle]: I swear I don't have OCD but you should capitalize this

File: app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java [hunk 3]
```diff
@@ -24,6 +27,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         tabs.addOnTabSelectedListener(new TabListener());
 
         map = view.findViewById(R.id.map);
+        map.setMinimumTileDpi(160);
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java]: Where did 160 come from?

**Comment by bstrauch24** [comment]
Does swiping around inside the images conflict with swiping between the three tabs?

File: app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java (hunk 3 above)
Comment (patrickfeltes) [reply app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java]: It's arbitrary. A comment in the source of the library said that 160-240 should be acceptable and would make images load faster. By default it is something in the 300s, and I didn't see much quality degradation by switching to this. Maybe I could create a variable called MINIMUM_TILE_DPI or something? What do you think?

**Comment by patrickfeltes** [comment]
> Does swiping around inside the images conflict with swiping between the three tabs?

I don't think there is a swipe gesture set up right now for this page.

File: app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java (hunk 3 above)
Comment (bstrauch24) [reply app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java]: sounds good to me, +1 for constant

File: app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java [hunk 4]
```diff
@@ -8,14 +8,16 @@
...
+import com.davemorrissey.labs.subscaleview.ImageSource;
+import com.davemorrissey.labs.subscaleview.SubsamplingScaleImageView;
+
 import org.hackillinois.android.R;
 
 public class IndoorMapsFragment extends Fragment {
-    private static final int DCL = 0;
-    private static final int ECEB = 1;
-    private static final int SIEBEL = 2;
+    private static final int[] imageResources =
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java]: YES!

File: app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java [hunk 5]
```diff
@@ -24,37 +26,23 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
-                map.setImageResource(R.drawable.dcl_indoor);
-                break;
-            case ECEB:
-                map.setImageResource(R.drawable.eceb_indoor);
-                break;
-            case SIEBEL:
-                map.setImageResource(R.drawable.siebel_indoor);
-                break;
-        }
+        map.setImage(ImageSource.resource(imageResources[index]));
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java]: So clean! :tada:

Comment (PradyumnaShome) [reply app/src/main/java/org/hackillinois/android/view/IndoorMapsFragment.java]: > Good programmers write good code. Great programmers write no code. Zen programmers delete code.

**Comment by patrickfeltes** [comment]
Adding new commits seems to dismiss approvals now. @bstrauch24 sorry to bug you for another approval.

**Comment by patrickfeltes** [comment]
I just changed the settings so that this won't happen again!

**Pull request merged**